src/Examine/LuceneEngine/Providers/LuceneIndexer.cs

```diff
@@ -1898,12 +1919,44 @@ private IndexWriter WriterFactory(Directory d)
             if (d == null) throw new ArgumentNullException(nameof(d));
             var writer = new IndexWriter(d, IndexingAnalyzer, false, IndexWriter.MaxFieldLength.UNLIMITED);
 
-            // set the error logging one
-            writer.SetMergeScheduler(new ErrorLoggingConcurrentMergeScheduler(Name,
-                (s, e) => OnIndexingError(new IndexingErrorEventArgs(s, -1, e))));
+            if (d is ExamineDirectory examineDirectory)
+            {
+                if (examineDirectory.GetDeletionPolicy() != null)
+                {
+                    writer = new IndexWriter(d, IndexingAnalyzer, true, examineDirectory.GetDeletionPolicy(),
+                        IndexWriter.MaxFieldLength.UNLIMITED);
+                }
 
+                if (examineDirectory.GetMergeScheduler() != null)
+                {
+                    writer.SetMergeScheduler(examineDirectory.GetMergeScheduler());
+                }
+
+                if (examineDirectory.IsReadOnly)
+                {
+                    DocumentWriting += (sender, args) => { args.Cancel = true; };
```

Comment:
As above, we shouldn't subscribe to our own events

Reply:
will change that